app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -0,0 +1,312 @@
+module ManageIQ::Providers::Redhat::InfraManager::ApiIntegration
+
+  extend ActiveSupport::Concern
+
+  included do
+    process_api_features_support
+  end
+
+  def supported_features
+    @supported_features ||= supported_api_versions.collect{|version| self.class.api_features[version]}.flatten.uniq
+  end
+
+  def connect(options = {})
+    raise "no credentials defined" if self.missing_credentials?(options[:auth_type])
+    version = options[:version] || 3
+    unless options[:skip_supported_api_validation] || supports_api_version?(version)
+      raise "version #{version} of the api is not supported by the provider"
+    end
+    # If there is API path stored in the endpoints table and use it:
+    path = options[:path] || default_endpoint.path
+    _log.info("Using stored API path '#{path}'.") unless path.blank?
+
+    server   = options[:ip] || address
+    port     = options[:port] || self.port
+    username = options[:user] || authentication_userid(options[:auth_type])
+    password = options[:pass] || authentication_password(options[:auth_type])
+    service  = options[:service] || "Service"
+
+    # Create the underlying connection according to the version of the oVirt API requested by
+    # the caller:
+    connect_method = "raw_connect_v#{version}".to_sym
+    connection = self.class.public_send(connect_method, server, port, path, username, password, service)
+
+    # Copy the API path to the endpoints table:
+    default_endpoint.path = version == 4 ? '/ovirt-engine/api' : connection.api_path
+
+    connection
+  end
+
+  def supported_api_versions
+    supported_api_versions_from_cache
+  end
+
+  def supported_api_versions_from_cache
+    Cacher.new(cache_key).fetch_fresh(last_refresh_date) { supported_api_verions_from_sdk }
+  end
+
+  def cache_key
+    "REDHAT_EMS_CACHE_KEY_#{id}"
```

Comment:
We can, but in the source code all the code related to oVirt is under a folder and name space named "redhat" that is why I used REDHAT. Considering this do you still think it should be OVIRT?